### PR TITLE
Fix the CustomNonLinearClsHead when the batch_size is set to 1

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

This PR introduces a new trick to avoid an error that happens in the bs=1 case at the `BN1d` included at the `CustomNonLinearClsHead`. For now, just applying to the `loss != IBLoss`, need to consider further cases.

### How to test

`otx train src/otx/algorithms/classification/configs/mobilenet_v3_large_1_cls_incr/template.yaml --train-data-roots tests/assets/classification_dataset --val-data-roots tests/assets/classification_dataset params --learning_parameters.batch_size 1`

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — COMMENTED** [review]
It just a question. It seems that error is raised due to dimension problem, is it right?
if it's right, how about changing dimension of tensor instead of concatenating same tensor?

**Comment by sungmanc** [comment]
> It just a question. It seems that error is raised due to dimension problem, is it right? if it's right, how about changing dimension of tensor instead of concatenating same tensor?

Right, however, if you mean just changing the shape of the tensor (i.e. [1, 960] --> [2, 480]. it could affect to the output since we used `BatchNorm1d`

**Comment by sungmanc** [comment]
Discard this PR and will made PR for the release branch

**Comment by eunwoosh** [comment]
> Right, however, if you mean just changing the shape of the tensor (i.e. [1, 960] --> [2, 480]. it could affect to the output since we used `BatchNorm1d`

ah I thought that [960] needs to be changed to [1, 960], but it's already [1, 960]. I think I just misunderstood it.
